baselines/fedrep/fedrep/dataset_preparation.py

```diff
@@ -0,0 +1,201 @@
+
+    def __getitem__(self, index):
+        """Get the item at the given index."""
+        data, targets = self.data[index], self.targets[index]
+        data_transform = (
+            self.train_data_transform if self.training else self.test_data_transform
+        )
+        if data_transform is not None:
+            data = data_transform(data)
+        return data, targets
+
+    def __len__(self):
+        """Return the length of the dataset."""
+        return len(self.targets)
+
+
+class CIFAR10(BaseDataset):
+    """CIFAR10 dataset."""
+
+    def __init__(
+        self,
+        root: Path = Path("datasets/cifar10"),
+        train_data_transform: transforms.transforms.Compose = None,
+        test_data_transform: transforms.transforms.Compose = None,
+    ):
+        super().__init__()
+
+        train_part = torchvision.datasets.CIFAR10(root, True, download=True)
+        test_part = torchvision.datasets.CIFAR10(root, False, download=True)
+        train_data = torch.tensor(train_part.data).permute([0, -1, 1, 2]).float()
+        test_data = torch.tensor(test_part.data).permute([0, -1, 1, 2]).float()
+        train_targets = torch.tensor(train_part.targets).long().squeeze()
+        test_targets = torch.tensor(test_part.targets).long().squeeze()
+        self.data = torch.cat([train_data, test_data])
+        self.targets = torch.cat([train_targets, test_targets])
+        self.classes = train_part.classes
+        self.train_data_transform = train_data_transform
+        self.test_data_transform = test_data_transform
+
+
+class CIFAR100(BaseDataset):
+    """CIFAR100 dataset."""
+
+    def __init__(
+        self,
+        root: Path = Path("datasets/cifar100"),
+        train_data_transform: transforms.transforms.Compose = None,
+        test_data_transform: transforms.transforms.Compose = None,
+    ):
+        super().__init__()
+        train_part = torchvision.datasets.CIFAR100(root, True, download=True)
+        test_part = torchvision.datasets.CIFAR100(root, False, download=True)
+        train_data = torch.tensor(train_part.data).permute([0, -1, 1, 2]).float()
+        test_data = torch.tensor(test_part.data).permute([0, -1, 1, 2]).float()
+        train_targets = torch.tensor(train_part.targets).long().squeeze()
+        test_targets = torch.tensor(test_part.targets).long().squeeze()
+        self.data = torch.cat([train_data, test_data])
+        self.targets = torch.cat([train_targets, test_targets])
+        self.classes = train_part.classes
+        self.train_data_transform = train_data_transform
+        self.test_data_transform = test_data_transform
+
+
+def call_dataset(dataset_name, root, **kwargs):
+    """Call the dataset."""
+    if dataset_name == "cifar10":
+        return CIFAR10(root, **kwargs)
+    if dataset_name == "cifar100":
+        return CIFAR100(root, **kwargs)
+    raise ValueError(f"Dataset {dataset_name} not supported.")
+
+
+# pylint: disable=R0914
+def randomly_assign_classes(
```

Comment:
This sounds to me like a pathological partitioning. Am I right? Could we replace this function (and likely parts o `dataset.dataset_main()`) with the [Flower Datasets](https://flower.ai/docs/datasets/ref-api/flwr_datasets.partitioner.PathologicalPartitioner.html#flwr_datasets.partitioner.PathologicalPartitioner) was of using the `PathologicalPartitioner`? 

Please take a look, I think `flwr-dataset` is quite powerful. Some of the examples (e.g.[examples/quickstart-pytorch](https://github.com/adap/flower/tree/main/examples/quickstart-pytorch)) show how to integrate them. You could follow a similar style as in the `client_fn` shown in that example.

Reply:
We are in the process of updating all of our examples to follow the same formate as `quickstart-pytorch`, `quickstart-tensorflow` and `quickstart-mlx` currently show: they run with `flwr run`. And a bit later we'll update the baselines too. All baselines will be updated to use `flwr-datasets` 👼

Reply:
Sure. I will try `flwr-dataset` later this week and think about the refactor. It may takes some times. 
Hope you don' mind.

Reply:
Also, please consider review my another PR #3663 .
Happy to see any comments and suggestions on that.